src/openshift/service.ts

```diff
@@ -51,24 +51,18 @@ export class Service extends OpenShiftItem {
     }
 
     static async del(treeItem: OpenShiftObject): Promise<string> {
-        let project: OpenShiftObject;
-        let application: OpenShiftObject;
-        let service: OpenShiftObject = treeItem;
-        if (!service) {
-            project = await vscode.window.showQuickPick(Service.odo.getProjects(), {placeHolder: "From which Project you want to delete Service"});
-            application = await vscode.window.showQuickPick(Service.odo.getApplications(project), {placeHolder: "From which Application you want to delete Service"});
-            if (application) {
-                service = await vscode.window.showQuickPick(Service.odo.getServices(application), {placeHolder: "Select Service to delete"});
-            }
-        }
+        let service: OpenShiftObject;
+        service = await Service.getOpenShiftCmdData(treeItem,
+            "From which Project you want to delete Service",
+            "From which Application you want to delete Service"
+        );
+        if (!treeItem) service = await vscode.window.showQuickPick(Service.getServiceNames(service), {placeHolder: "Select Service to delete"});
```

Comment:
```if (!application) service = await vscode.window.showQuickPick(Service.getServiceNames(application), {placeHolder: "Select Service to delete"});```

Reply:
`!application` always return false

Reply:
After 
``` TypeScript
Service.getOpenShiftCmdData(treeItem, 
    "From which Project you want to delete Service",
    "From which Application you want to delete Service"
);
```
`application` variable would be ether the same `treeItem` or application instance if `treeItem` is `null | undefined` or `null` if quickpick was canceled.

Reply:
Your code would probably work, I have not tested it, but it just look confusing to me at first sight.